#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_sigpipe.dpatch based on patch by <james.westby@canonical.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Restore the default SIGPIPE action so that if a parent blocks
## DP: SIGPIPE (e.g. gksu) then we still receive it from capture children,
## DP: and so don't hang when the capture preview window is closed.

@DPATCH@
--- trunk-ref/dumpcap.c	(revision 28548)
+++ trunk/dumpcap.c	(revision 28549)
@@ -2629,6 +2618,7 @@
   sigemptyset(&action.sa_mask);
   sigaction(SIGTERM, &action, NULL);
   sigaction(SIGINT, &action, NULL);
+  sigaction(SIGPIPE, &action, NULL);
   sigaction(SIGHUP, NULL, &oldaction);
   if (oldaction.sa_handler == SIG_DFL)
     sigaction(SIGHUP, &action, NULL);
